extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/DevModeExecutorService.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.vertx.core.runtime;
+
+import java.util.concurrent.ExecutorService;
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+
+/**
+ * This executor is only used in the dev mode as the Vertx worker thread pool.
+ * <p>
+ * The underlying executor can be shut down and then replaced with a new re-initialized executor.
+ */
+class DevModeExecutorService extends ForwardingExecutorService {
+
+    private static final Logger LOG = Logger.getLogger(DevModeExecutorService.class);
+
+    private final Supplier<ExecutorService> initializer;
+    private volatile ExecutorService executor;
+
+    DevModeExecutorService(Supplier<ExecutorService> initializer) {
+        this.initializer = initializer;
+        this.executor = initializer.get();
```

Comment:
It would be nice to add an `Objects.requireNonNull()` here for safety check purposes

Reply:
It's a package-private class so I slacked off a bit :D